src/lib/comms/sol-oic-cbor.c

```diff
@@ -63,7 +63,7 @@ initialize_cbor_payload(struct sol_oic_map_writer *encoder)
 
     cbor_encoder_init(&encoder->encoder, encoder->payload,
         buf->capacity - offset, 0);
-    old_ptr = encoder->encoder.ptr;
+    old_ptr = encoder->encoder.data.ptr;
```

Comment:
I recall well (it's been a while now), the memory region we pass to tinycbor to encode is a Soletta buffer, that is memory that can be realloc-ed if it grows too much. That hack is to hard-wire into an existing cbor encoder the new memory region with its size and let the encoding proceed without noticing that. At the time it was written, there was no API supporting that mem region changing on the fly. Not sure if there is one already on this last release. From what I checked as well, this data field does not exist indeed (only checked now). So not sure what you wanted.